OracleDatabase/SingleInstance/dockerfiles/18.4.0/runOracle.sh

```diff
@@ -0,0 +1,218 @@
+   fi;
+
+   cp $ORACLE_BASE/oradata/dbconfig/$ORACLE_SID/oratab /etc/oratab
+}
+
+########### SIGTERM handler ############
+function _term() {
+   echo "Stopping container."
+   echo "SIGTERM received, shutting down database!"
+  /etc/init.d/oracle-xe-18c stop
+}
+
+########### SIGKILL handler ############
+function _kill() {
+   echo "SIGKILL received, shutting down database!"
+   /etc/init.d/oracle-xe-18c stop
+}
+
+############# Create DB ################
+function createDB {
+   # Auto generate ORACLE PWD if not passed on
+   export ORACLE_PWD=${ORACLE_PWD:-"`openssl rand -hex 8`"}
+   echo "ORACLE PASSWORD FOR SYS AND SYSTEM: $ORACLE_PWD";
+
+   (echo "$ORACLE_PWD"; echo "$ORACLE_PWD";) | /etc/init.d/oracle-xe-18c configure
+
+   # Listener 
+   echo "# listener.ora Network Configuration File:
+         
+         SID_LIST_LISTENER = 
+           (SID_LIST =
+             (SID_DESC =
+               (SID_NAME = PLSExtProc)
+               (ORACLE_HOME = $ORACLE_HOME)
+               (PROGRAM = extproc)
+             )
+           )
+         
+         LISTENER =
+           (DESCRIPTION_LIST =
+             (DESCRIPTION =
+               (ADDRESS = (PROTOCOL = IPC)(KEY = EXTPROC_FOR_XE))
+               (ADDRESS = (PROTOCOL = TCP)(HOST = 0.0.0.0)(PORT = 1521))
+             )
+           )
+         
+         DEFAULT_SERVICE_LISTENER = (XE)" > $ORACLE_HOME/network/admin/listener.ora
+
+# TNS Names.ora
+   echo "# tnsnames.ora Network Configuration File:
+
+XE =
+  (DESCRIPTION =
+    (ADDRESS = (PROTOCOL = TCP)(HOST = 0.0.0.0)(PORT = 1521))
+    (CONNECT_DATA =
+      (SERVER = DEDICATED)
+      (SERVICE_NAME = XE)
+    )
+  )
+
+LISTENER_XE =
+  (ADDRESS = (PROTOCOL = TCP)(HOST = 0.0.0.0)(PORT = 1521))
+
+XEPDB1 =
+  (DESCRIPTION =
+    (ADDRESS = (PROTOCOL = TCP)(HOST = 0.0.0.0)(PORT = 1521))
+    (CONNECT_DATA =
+      (SERVER = DEDICATED)
+      (SERVICE_NAME = XEPDB1)
+    )
+  )
+
+EXTPROC_CONNECTION_DATA =
+  (DESCRIPTION =
+     (ADDRESS_LIST =
+       (ADDRESS = (PROTOCOL = IPC)(KEY = EXTPROC_FOR_XE))
+     )
+     (CONNECT_DATA =
+       (SID = PLSExtProc)
+       (PRESENTATION = RO)
+     )
+  )
+" > $ORACLE_HOME/network/admin/tnsnames.ora
+
+  # Move database operational files to oradata
+  moveFiles;
+}
+
+############# MAIN ################
+
+# Set SIGTERM handler
+trap _term SIGTERM
+
+# Set SIGKILL handler
+trap _kill SIGKILL
+
+# Check whether database already exists
+if [ -d $ORACLE_BASE/oradata/$ORACLE_SID ]; then
+   symLinkFiles;
+   # Make sure audit file destination exists
+   if [ ! -d $ORACLE_BASE/admin/$ORACLE_SID/adump ]; then
+      su -p oracle -c "mkdir -p $ORACLE_BASE/admin/$ORACLE_SID/adump"
+   fi;
+fi;
+
+/etc/init.d/oracle-xe-18c start | grep -qc "Oracle Database is not configured"
+if [ "$?" == "0" ]; then
+   # Check whether container has enough memory
+   if [ `df -Pk /dev/shm | tail -n 1 | awk '{print $2}'` -lt 1048576 ]; then
```

Comment:
Shared memory is no longer needed in 18c as we do not use Automatic Memory Management. The whole `/dev/shm` if clause can go away.